### ISSUE TITLE
BUG: null pointer dereference after cancelling a `close()`

### ISSUE THREAD

**Issue by Xiangze-Li** [ctabin/libzippp#220]
In the function `static void callErrorHandlingCallback(zip* zipHandle, const std::string& msg, ErrorHandlerCallback* callback)`, it says:

```cpp
if(zipHandle!=nullptr) {
    zip_error_t* error_code = zip_get_error(zipHandle);
    callErrorHandlingCallbackFunc(error_code->str, error_code->zip_err, error_code->sys_err, callback);
} else {
    // ...
}
```

However, `error_code->str` is nullable, said in libzip as follows:

```c
/* error information */
/* use zip_error_*() to access */
struct zip_error {
    int zip_err;         /* libzip error code (ZIP_ER_*) */
    int sys_err;         /* copy of errno (E*) or zlib error code */
    char *_Nullable str; /* string representation or NULL */
};
```

When a progress callback cancels the `close` process, the `zip_error` does have a NULL `str` field, which causes a null pointer deref.

**Comment by ctabin** [comment]
Thanks for the report and the PR.
Fixed by #221.

**Comment by fdgStilla** [comment]
@ctabin Could you create a release including this fix ? We cannot use the last version because of this crash.
